esp/services/ws_workunits/ws_workunitsService.hpp

```diff
@@ -361,6 +362,12 @@ class CWsWorkunitsSoapBindingEx : public CWsWorkunitsSoapBinding
         VStringBuffer xpath("Software/EspProcess[@name=\"%s\"]/EspBinding[@name=\"%s\"]/BatchWatch", process, name);
         batchWatchFeaturesOnly = cfg->getPropBool(xpath.str(), false);
         directories.set(cfg->queryPropTree("Software/Directories"));
+
+        xpath.setf("Software/EspProcess[@name=\"%s\"]/EspBinding[@name=\"%s\"]/service", process, name);
```

Comment:
not for now: really anything like this that builds up an xpath to probe the environment or similar, should move to somewhere common, e.g. into environment.*, e.g. IPropertyTree *getEnvServiceBinding(process, serviceName)

Reply:
I need both service and binding here. I am going to create:
IPropertyTree *getEnvServiceBinding(process, bindingName);
IPropertyTree *getEnvServiceBinding(process, serviceName);

Reply:
>the xpath here is for reading esp.xml

ok, I didn't realize that, it looks like a clone of the structures from environment.xml sections.

Either way, I would look at this after this separately, after this PR is complete.
There are probably other places that would similarly benefit from moving the manually created xpaths to helper methods in environment.*

@wangkx